### PR TITLE
add version to app.py routes

### PR DESCRIPTION
**Pull request by miararoy**
## Problem

Versioning an API is a crucial aspect of maintaining its usability and compatibility as it evolves over time. There are several approaches to API versioning but the most common is URI Versioning (Path Versioning): where the version number is included in the URL path. For example: 

```
GET /v1/items
POST /v2/items
```

## Solution

- added API_VERSION const to app.py (we could think about a better place to manage this in the future
- changed all routes
- removed /context from chat/completions

> Note:  did not change /health and /shutdown

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Infrastructure change (CI configs, etc)
- [ ] Non-code change (docs, etc)
- [ ] None of the above: (explain here)

## Test Plan

Regression

### CODE REVIEW

File: src/canopy_server/app.py
Comment (igiloh-pinecone) [inline src/canopy_server/app.py:1]: @miararoy You need to change the default `url` paths of all CLI commands that talk to the server (`canopy chat`, `canopy health` etc).  
Please do so after #170 is merged, otherwise you'll have conflicts.

File: src/canopy_server/app.py
Comment (igiloh-pinecone) [inline src/canopy_server/app.py:86]: Again, you need to change that in `cli.py` as well (but wait for #170 to merge first!)

File: src/canopy_server/app.py
Comment (igiloh-pinecone) [inline src/canopy_server/app.py:88]: Personally I find this `/api` redundant.  
Up to you, but I vote against it.

File: tests/e2e/test_app.py
Comment (igiloh-pinecone) [inline tests/e2e/test_app.py:99]: That kinda misses the point... 

You want to pass a `base_url` arg to `TestClient()`, and leave the test cases themselves untouched.

File: src/canopy_server/app.py
Comment (igiloh-pinecone) [inline src/canopy_server/app.py:192]: @miararoy if you're adding versioning - than **all** API endpoints should have version, including `/health` etc (it has a structured return value, that may change between versions)

**Review by igiloh-pinecone — CHANGES_REQUESTED** [review]
@miararoy see a few suggestions

File: src/canopy_server/app.py
Comment (miararoy) [reply src/canopy_server/app.py:203]: done

File: tests/e2e/test_app.py
Comment (miararoy) [reply tests/e2e/test_app.py]: fixed

File: src/canopy_server/app.py
Comment (miararoy) [reply src/canopy_server/app.py]: removed

File: src/canopy_server/app.py
Comment (miararoy) [reply src/canopy_server/app.py:86]: Think im' done, resolving this for the future review

File: src/canopy_cli/cli.py
Comment (igiloh-pinecone) [inline src/canopy_cli/cli.py:585]: Let's define a `DEFAULT_URL` at the top of the file?  
It suddenly makes much more sense

Comment (miararoy) [reply src/canopy_cli/cli.py]: done

File: src/canopy_cli/cli.py
Comment (igiloh-pinecone) [inline src/canopy_cli/cli.py:533]: This particular change wasn't tested on all OSes (passing `host='localhost'` to `uvicorn.run()`).  
If you wanna merge it, we will need to test it.

Comment (miararoy) [reply src/canopy_cli/cli.py:533]: does 0.0.0.0 works?

Comment (miararoy) [reply src/canopy_cli/cli.py:533]: on all OSs - that is

File: src/canopy_server/models/v1/api_models.py
Comment (igiloh-pinecone) [inline src/canopy_server/models/v1/api_models.py:7]: 👍

**Review by igiloh-pinecone — APPROVED** [review]
@miararoy LGTM, but see [this comment](https://github.com/pinecone-io/canopy/pull/169/files#r1392634940) before merging

File: src/canopy_cli/cli.py
Comment (igiloh-pinecone) [reply src/canopy_cli/cli.py:533]: If that was the code in `main` before this PR - then yeah. I tested this code on all 3 OSes manually yesterday.

**Pull request merged**
